qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -0,0 +1,557 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Stabilizer state class.
+"""
+
+import copy
+import numpy as np
+
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit.instruction import Instruction
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info.states.quantum_state import QuantumState
+from qiskit.quantum_info.operators.symplectic import Clifford, Pauli
+from qiskit.quantum_info.operators.symplectic.clifford_circuits import _append_x
+
+
+class StabilizerState(QuantumState):
+    """StabilizerState class. Based on the internal class:
```

Comment:
Since this is a public-facing class, I think some explanations are in order: What is this class? How it differs from `Clifford`? What representation are we using? The AG representation is more complex than the basic textbook description of a Clifford element; it has destabilizers as well as stabilizers, and the specific tableau structure given in AG. This should be explicitly mentioned here.

Reply:
I added an example.